### ISSUE TITLE
add support for fragmented slides.

### ISSUE THREAD

**Issue by isaacabraham** [fsprojects/FsReveal#36]

**Comment by forki** [comment]
Can you please give more details?

**Comment by isaacabraham** [comment]
http://lab.hakim.se/reveal-js/#/fragments

**Comment by reidev275** [comment]
maybe not ideal, but if you're using markdown you can input the html directly

<section>
<p class="fragment roll-in">Foo</p>
<p class="fragment grow">Bar</p>
</section>

**Comment by forki** [comment]
Please propose a markdown syntax for:

  <section>
  	<section id="fragments">
  		<h2>Fragments</h2>
  		<p>Hit the next arrow...</p>
  		<p class="fragment">... to step through ...</p>
  		<p><span class="fragment">... a</span> <span class="fragment">fragmented</span> <span class="fragment">slide.</span></p>
  
  		<aside class="notes">
  			This slide has fragments which are also stepped through in the notes window.
  		</aside>
  	</section>
  	<section>
  		<h2>Fragment Styles</h2>
  		<p>There's different types of fragments, like:</p>
  		<p class="fragment grow">grow</p>
  		<p class="fragment shrink">shrink</p>
  		<p class="fragment roll-in">roll-in</p>
  		<p class="fragment fade-out">fade-out</p>
  		<p class="fragment current-visible">current-visible</p>
  		<p class="fragment highlight-red">highlight-red</p>
  		<p class="fragment highlight-blue">highlight-blue</p>
  	</section>
  </section>

**Comment by forki** [comment]
ping.

we still need an idea for the syntax.

**Comment by kimsk** [comment]
Could we do something similar to what `reveal.js` [does](https://github.com/hakimel/reveal.js#element-attributes)?

I am thinking about something like this:
    ## Fragment Styles
    There's different types of fragments, like:
    grow 
    shrink

**Comment by reidev275** [comment]
The issue seems to be, "how do we add html attribute syntax to markdown?"

While I think it's probably as easy to just write the html (same issue with all other approaches of adding attributes that I've seen), I think @kimsk 's idea to continue the syntax of reveal.js would keep the two projects in sync which would probably have more benefits than rolling anything custom unless a really clean syntax is proposed.

**Comment by reidev275** [comment]
Probably a dumb question, but here goes...,
I have what I think is a working example but I'm not sure how to take the code I have in my branch and use it to actually build a site from markdown.  How do you all do it?

**Comment by forki** [comment]
You mean building slides? Did you clone the master branch and run the
build.cmd? It should open a browser.

**Comment by CumpsD** [comment]
Have you looked at https://github.com/hakimel/reveal.js/pull/677?

They included a reference too: https://github.com/VonC/reveal.js/blob/classes_in_markdown/test/test-element-attributes-markdown.html

```
## Slide 1.1 {_class="fragment fade-out" data-fragment-index="1"}

						--

						## Slide 1.2 {_class="fragment shrink"}

						Paragraph 1 {_class="fragment grow"}

						Paragraph 2 {_class="fragment grow"}

						- list item 1  {_class="fragment roll-in"}
						- list item 2  {_class="fragment roll-in"}
						- list item 3  {_class="fragment roll-in"}


						---

						## Slide 2


						Paragraph 1.2  
						multi-line {_class="fragment highlight-red"}

						Paragraph 2.2 {_class="fragment highlight-red"}

						Paragraph 2.3 {_class="fragment highlight-red"}

						Paragraph 2.4 {_class="fragment highlight-red"}

						- list item 1  {_class="fragment highlight-green"}
						- list item 2  {_class="fragment highlight-green"}
						- list item 3  {_class="fragment highlight-green"}
						- list item 4  {_class="fragment highlight-green"}
						- list item 5  {_class="fragment highlight-green"}
            
Test
            
![Example Picture{_class="reveal stretch"}](red-curtain-50x50.jpg)
```

**Comment by reidev275** [comment]
@forki Indeed, from the master branch I can run build.cmd and it opens the browser, but from the master branch I'm not sure how to edit the FsReveal source.  I branched off of develop to make code changes.  Perhaps that's the source of my error?

**Comment by kimsk** [comment]
Are you talking about editing FsReveal code (FsReveal.dll)?

**Comment by reidev275** [comment]
Yes, mapping paragraph literals with the pattern you mentioned into literals with html expected from reveal

**Comment by forki** [comment]
Ah now I get the issue. You want to test your custom built fsreveal.dll!?

There are two simple ways:
1) create an unittest (there are samples)
or 2) clone master branch into a different folder and set the build output
of the develop branch to overwrite the fsreveal.dll in the master branch
folder.
